### PR TITLE
Backport commits from firecracker repository

### PR DESCRIPTION
**Pull request by 0x00A5**
Signed-off-by: YUAN LYU <lyuyuan92@gmail.com>

## Reason for This PR

As part of the requirements of resolving [firecracker issue 1670](https://github.com/firecracker-microvm/firecracker/issues/1670), I am opening this PR to even up code changes between firecracker and micro-http repo.

## Description of Changes

* Put latest `src` in micro-http repo
* Removed `logger` dependency per [this comment](https://github.com/firecracker-microvm/firecracker/issues/1670#issuecomment-775029893)

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.

### CODE REVIEW

**Comment by serban300** [comment]
@lyuyuan Thanks for the contribution ! Could you please cherry pick the actual commits ? I think the PR would be easier to review this way.

**Comment by 0x00A5** [comment]
Hi @serban300 , sorry for the delay, I finally found time to work on this PR again. I have cherry picked all commits and committed some cleanup to resolve errors which might be caused by mistakes I made while merging.
Please take a look, thank you!

**Comment by serban300** [comment]
@lyuyuan The changes look good, but I see that the coverage has dropped by 1.6%. This is pretty much. Probably we didn't notice it in Firecracker since those code paths are covered by other tests from the vmm.

Usually we aim to keep the coverage as least as high as it was before in order to merge a PR. I'm not sure if we should treat the current situation as an exception since this is how the code looks like in Firecracker. @firecracker-microvm/compute-capsule what do you think ?

**Comment by 0x00A5** [comment]
@serban300 I added a few unit tests to cover some red lines shown in `kcov` report. Please take a look.

**Review by acatangiu — DISMISSED** [review]

**Comment by serban300** [comment]
@lyuyuan Could you also add your sign off to the commits that are ported from firecracker please ? Otherwise the changes look good. Thanks !

**Comment by 0x00A5** [comment]
@serban300 @acatangiu I just added my signoff to the last 26 commits on the branch. Please take a look, thanks!

File: Cargo.toml [hunk 1]
```diff
@@ -6,4 +6,5 @@ authors = ["Amazon Firecracker team <firecracker-devel@amazon.com>"]
 edition = "2018"
 
 [dependencies]
-epoll = ">=4.0.1"
+libc = ">=0.2.39"
+vmm-sys-util = ">=0.6.1"
```
Comment (serban300) [inline Cargo.toml:10]: Nit: should add new line

**Review by serban300 — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

**Pull request merged**